LdapInterop/UserSynchronizer.php

```diff
@@ -197,6 +197,43 @@ public function synchronizePiwikAccessFromLdap($piwikLogin, $ldapUser)
         }
     }
 
+    /**
+     * Purges a supplied Matomo login from DB based on LDAP entity absence.
+     *
+     * @param string $matomoLogin The username of the user who will be purged.
+     * @return string The Matomo user login that was purged.
+     */
+    public function purgeLdapUser($matomoLogin)
+    {
+        $userMapper = $this->userMapper;
+        $usersManagerAPI = $this->usersManagerApi;
+        $userModel = $this->userModel;
+        $logger = $this->logger;
+        return Access::doAsSuperUser(function () use ($matomoLogin, $userMapper, $usersManagerAPI, $userModel, $logger) {
+            $matomoLogin = $userMapper->getExpectedLdapUsername($matomoLogin);
+
+            $existingUser = $userModel->getUser($matomoLogin);
+
+            $logger->debug("UserSynchronizer::{func}: purging user [ Matomo login = {matomoLogin} ]", array(
+                'func' => 'purgeLdapUser',
+                'matomoLogin' => $matomoLogin
+            ));
+
+            if (empty($existingUser)) {
+                $logger->warning("Unable to purge LDAP user '{user}': no such user.", array('user' => $matomoLogin));
+                return FALSE;
+            } else {
+                if (!$userMapper->isUserLdapUser($existingUser['login'])) {
```

Comment:
Can you also do this check in the synchronize users command? This will remove superfluous output for the command.

Reply:
[`purgeUser` in `API.php`](https://github.com/matomo-org/plugin-LoginLdap/pull/205/files#diff-b9905f307279d5bb39c664e2a57b849eR158) already makes sure users being purged are LDAP ones.
This warning is a failsafe, and shall never be triggered. If it is, it would be a sign of trouble.